### PR TITLE
Implement MeshcoreBackboneNode for backbone layer communication

### PR DESCRIPTION
**Pull request by loredous**
## Summary

This PR implements the MeshcoreBackboneNode, completing the backbone layer of the OmniMesh three-layer architecture. With this implementation, OmniMesh bridges can now communicate with each other over the Meshcore mesh network.

### Key Features

- **Multi-connection support**: Serial, TCP, and BLE connections to Meshcore devices
- **Async message handling**: Uses Meshcore's EventDispatcher pattern with native async/await
- **Automatic bridge discovery**: Learns bridge ID to public key mappings from incoming messages
- **Flexible addressing**: Direct send to known bridges or broadcast to all contacts
- **Reliable connectivity**: Auto-reconnect with configurable retry attempts

### Design Decisions

- **Base64 encoding**: BackboneMessages are base64 encoded for transport since Meshcore uses text-based messages
- **Broadcast fallback**: Messages to unknown bridges are broadcast to all contacts
- **Pattern consistency**: Follows the same implementation patterns as MeshtasticAccessNode

### Files Changed

| File | Description |
|------|-------------|
| `src/omnimesh/nodes/meshcore/__init__.py` | Package init |
| `src/omnimesh/nodes/meshcore/node.py` | MeshcoreBackboneNode implementation |
| `src/tests/unit/test_meshcore_node.py` | 27 comprehensive unit tests |
| `CLAUDE.md` | Updated project status and documentation |

### Testing

- 27 new unit tests covering initialization, lifecycle, message sending/receiving, and bridge mapping
- All 86 tests in the test suite pass

## Test plan

- [ ] Review code for correctness and adherence to project patterns
- [ ] Verify unit tests cover edge cases
- [ ] Test with actual Meshcore hardware (if available)
- [ ] Integration test with MeshtasticAccessNode and OmnimeshBridge

🤖 Generated with [Claude Code](https://claude.ai/claude-code)

### CODE REVIEW

File: src/omnimesh/nodes/meshcore/node.py [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+                    msg=encoded_message,
+                    timestamp=int(time.time()),
+                )
+                self._logger.debug(
+                    f"Broadcast sent to {contact_info.get('adv_name', pubkey[:16])}"
+                )
+            except Exception as e:
+                self._logger.error(
+                    f"Failed to send to contact {pubkey[:16]}...: {e}"
+                )
```
Comment (loredous) [inline src/omnimesh/nodes/meshcore/node.py:324]: This should be re-implemented to send the data to a channel, not to specific contacts

**Comment by loredous** [comment]
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/loredous/OmniMesh/pull/6#pullrequestreview-3733082910)

**Comment by loredous** [comment]
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/loredous/OmniMesh/pull/6#pullrequestreview-3733150344)
